pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -189,6 +196,7 @@ def __init__(
         mode: str = "min",
         auto_insert_metric_name: bool = True,
         every_n_train_steps: Optional[int] = None,
+        train_time_interval: Optional[timedelta] = None,
```

Comment:
it's intended for time between training_step calls. you're right that this could include validation if someone runs validation during the middle of the training epoch. i added `train` also to keep the pattern of `every_n_train_steps` / `every_n_val_epochs` which mention which running stage they're called in